src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,617 @@
+        await self.get_cluster_from_type(uint)
+
+        # Step 12
+        # TH sends the Read Request Message to the DUT to read an attribute of data type signed integer.
+        # If the device does not have an attribute of data type signed integer, skip this step.
+
+        # Verify on the TH that the DUT returns data from the expected attribute path.
+        self.print_step(12, "Read a read request of attribute type signed integer")
+        await self.get_cluster_from_type(int)
+
+        # Step 13
+        # TH sends the Read Request Message to the DUT to read an attribute of data type floating point.
+        # If the device does not have an attribute of data type floating point, skip this step.
+
+        # Verify on the TH that the DUT returns data from the expected attribute path.
+        self.print_step(13, "Read a read request of attribute type floating point")
+        await self.get_cluster_from_type(float)
+
+        # Step 14
+        # TH sends the Read Request Message to the DUT to read an attribute of data type Octet String.
+        # If the device does not have an attribute of data type octet string, skip this step.
+
+        # Verify on the TH that the DUT returns data from the expected attribute path.
+        self.print_step(14, "Read a read request of attribute type octet string")
+        await self.get_cluster_from_type(bytes)
+
+        # Step 15
+        # TH sends the Read Request Message to the DUT to read an attribute of data type Struct.
+        # If the device does not have an attribute of data type struct, skip this step.
+
+        # Verify on the TH that the DUT returns data from the expected attribute path.
+        self.print_step(15, "Read a read request of attribute type struct")
+        await self.get_cluster_from_type(ClusterObject)
+
+        # Step 16
+        # TH sends the Read Request Message to the DUT to read an attribute of data type List.
+        # If the device does not have an attribute of data type list, skip this step.
+        # Verify on the TH that the DUT returns data from the expected attribute path.
+        self.print_step(16, "Read a read request of attribute type list")
+        await self.get_cluster_from_type(list)
+
+        # Step 17
+        # TH sends the Read Request Message to the DUT to read an attribute of data type enum.
+        # If the device does not have an attribute of data type enum, skip this step.
+        # Verify on the TH that the DUT returns data from the expected attribute path.
+        self.print_step(17, "Read a read request of attribute type enum")
+        await self.get_cluster_from_type(MatterIntEnum)
+
+        # Step 18
+        # TH sends the Read Request Message to the DUT to read an attribute of data type bitmap.
+        # If the device does not have an attribute of data type bitmap, skip this step.
+        self.print_step(18, "Read a read request of attribute type bitmap")
+        await self.get_cluster_from_type(IntFlag)
```

Comment:
Does this also need special handling like MatterIntEnum? I don't see it in the function above, and it seems similar since these are derived.

Reply:
I think you might be right and I can add the checks tomorrow (August 9).